### ISSUE TITLE
WorkspaceSaga test case passing inconsistently

### ISSUE THREAD

**Issue by teekaytai** [source-academy/frontend#2080]
In the file `src/commons/sagas/__tests__/WorkspaceSaga.ts`, the test labelled "does not show infinite loop error if coinflip is false" currently passes or fails randomly even when no code is edited.

The error is "`TypeError: Cannot read properties of undefined (reading 'explain')`"

![image](https://user-images.githubusercontent.com/90845551/151703712-0650508d-e285-416f-a784-a2d91fa3ed3d.png)

**Comment by chownces** [comment]
Hi @teekaytai! I cant seem to reproduce this issue on my end. Could you check if you have the latest master branch and dependencies installed?

**Comment by teekaytai** [comment]
Hi! Just double checked by cloning the repo again and running `yarn install` and `yarn run format`. Running `yarn run test`, the test still fails some of the time.
Not sure if it helps but I'm on Windows and using WSL. My node version is the LTS version 16.13.2.

**Comment by ngxingyu** [comment]
I do see this test case fail occasionally as well, not entirely sure why but I'm using WSL as well, seems to occur when my CPU usage is higher
![image](https://user-images.githubusercontent.com/12682160/152189060-4db549a8-ee84-45e5-99c6-0448b1cb597f.png)

**Comment by lhw-1** [comment]
I occasionally get the same issue as well, on both Ubuntu 20.0.4 and Fedora 34:

![Screenshot from 2022-02-04 18-43-00](https://user-images.githubusercontent.com/25716529/152515917-6b3c27ee-3d28-4ba5-b3ea-b8b9f4aa240d.png)

It seems to randomly pass some of the times, and fail at other times.

**Comment by martin-henz** [comment]
Same here. @JoeyChenSmart can you look into this? Tx.

**Comment by JoeyChenSmart** [comment]
Noted, will take a look.
Side note: since we're not doing any 'experiments' with the infinite loop detector for now, should we remove the coinflip functionality (where we disable the detector 50% of the time)?

**Comment by martin-henz** [comment]
Yes, I think we have shown that the infinite loop detector is effective and has a positive impact on learning. We can remove the coin flip for now, until we resume experiments, perhaps for Sem 1 AY2022/23.
